presto-docs/src/main/sphinx/security/ldap.rst

```diff
@@ -0,0 +1,212 @@
+
+To enable LDAP authentication for Presto, configuration changes are made on
+the Presto coordinator. No changes are required to the worker configuration;
+only the communication from the clients to the coordinator is authenticated.
+
+Presto Server Configuration
+---------------------------
+
+Environment Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. _ldap_server:
+
+LDAP Server
+~~~~~~~~~~~
+
+The Presto clients send a user name and password to the coordinator and
+coordinator validates these credentials using an external LDAP service.
+The LDAP implementation types supported by Presto are Active Directory
+and OpenLDAP.
+
+We recommend to always use Secure LDAP (LDAPS) in production. So before 
+configuring Presto for LDAP Authentication, make sure that you have
+TLS enabled on your LDAP Server. You need to import the LDAP Server's
+TLS Certificate to the default Java truststore of the Presto coordinator
+to secure TLS connection. You can use the `keytool` command to
+import this certificate ``ldap_server.crt`` to the truststore
+on the coordinator.
+
+.. code-block:: none
+
+    $ keytool -import -keystore <JAVA_HOME>/jre/lib/security/cacerts -trustcacerts -alias ldap_server -file ldap_server.crt
+
+Java Keystore File for TLS
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using LDAP authentication, access to the Presto coordinator should be
+through HTTPS. You can do it by creating a :ref:`server_java_keystore` on
+the coordinator.
+
+Presto Coordinator Node Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You must make the following changes to the environment prior to configuring the
+Presto coordinator to use LDAP authentication and HTTPS. 
+
+ * :ref:`ldap_server`
+ * :ref:`server_java_keystore`
+
+config.properties
+~~~~~~~~~~~~~~~~~
+
+LDAP authentication is configured in the coordinator node's
+config.properties file. The entries that need to be added are listed below.
+
+.. code-block:: none
+
+    authentication.ldap.enabled=true
+    authentication.ldap.server-type=active_directory
+    authentication.ldap.url=ldaps://ldap-server:636
+    authentication.ldap.ad-domain=corp.domain.com
+
+    # Group based authorization properties (Optional)
+    authentication.ldap.base-dn=OU=Asia,DC=corp,DC=domain,DC=com
+    authentication.ldap.group-dn=CN=Group,OU=America,DC=corp,DC=domain,DC=com
+    authentication.ldap.user-object-class=person
+
+    http-server.https.enabled=true
+    http-server.https.port=8443
+
+    http-server.https.keystore.path=/etc/presto_keystore.jks
+    http-server.https.keystore.key=keystore_password
+
+======================================================= ======================================================
+Property                                                Description
+======================================================= ======================================================
+``authentication.ldap.enabled``                         Enable LDAP Authentication for the Presto coordinator.
```

Comment:
It might be a good idea to specify what are the default values for these properties.